src/main/java/org/opentripplanner/netex/mapping/FlexStopsMapper.java

```diff
@@ -59,85 +57,82 @@ class FlexStopsMapper {
    * dependent on a key/value in the NeTEx file, until proper NeTEx support is added.
    */
   StopLocation map(FlexibleStopPlace flexibleStopPlace) {
-    Object area = flexibleStopPlace
-      .getAreas()
-      .getFlexibleAreaOrFlexibleAreaRefOrHailAndRideArea()
-      .get(0);
-    if (!(area instanceof FlexibleArea)) {
-      issueStore.add(
-        Issue.issue(
-          "UnsupportedFlexibleStopPlaceAreaType",
-          "FlexibleStopPlace %s contains an unsupported area.",
-          flexibleStopPlace.getId()
-        )
-      );
-      LOG.warn(
-        "FlexibleStopPlace {} not mapped. Hail and ride areas are not currently supported.",
-        flexibleStopPlace.getId()
-      );
-      return null;
-    }
+    List<StopLocation> stops = new ArrayList<>();
 
-    Optional<KeyValueStructure> flexibleAreaType = Optional.empty();
-    if (flexibleStopPlace.getKeyList() != null) {
-      flexibleAreaType =
-        flexibleStopPlace
-          .getKeyList()
-          .getKeyValue()
-          .stream()
-          .filter(k -> k.getKey().equals(FLEXIBLE_STOP_AREA_TYPE_KEY))
-          .findFirst();
+    var areas = flexibleStopPlace.getAreas().getFlexibleAreaOrFlexibleAreaRefOrHailAndRideArea();
+    for (var area : areas) {
+      if (!(area instanceof FlexibleArea flexibleArea)) {
+        issueStore.add(
+          Issue.issue(
+            "UnsupportedFlexibleStopPlaceAreaType",
+            "FlexibleStopPlace %s contains an unsupported area %s.",
+            flexibleStopPlace.getId(),
+            area
+          )
+        );
+        LOG.warn(
+          "FlexibleStopPlace {} contains an unsupported area {}. Hail and ride areas are not currently supported.",
+          flexibleStopPlace.getId(),
+          area
+        );
```

Comment:
When an issue is added to the issue store it is logged with `debug` level. We could possibly change this, but I do not think we should log here. If we want to log different issues with different log level we should add extra methods to the `DataImportIssueStore` e.g. `addAndLogWarn`, `addAndLogInfo` and `addAndLogDebug`. Another but too me better solution is to inside the `DataImportIssueStore` to create a logger per issue type (o.o.issue.<issue-type>) and log as info. Then each deployment can control the logging as they like.

I can do this in another PR - just want to discuss it in a dev meeting first. In this PR i suggest just removing the log statment.

Reply:
Should we remove the other 2 LOG.warn statements as well?